test/gtest/uct/test_sockaddr.cc

```diff
@@ -1040,6 +1074,34 @@ UCS_TEST_P(test_uct_sockaddr, ep_disconnect_err_codes)
     }
 }
 
+
+// The idea of the test is to check that execution of a user's conn_request_cb
+// could be done simultaneously with uct_listener_destroy() calling from
+// another thread
+UCS_TEST_P(test_uct_sockaddr, concurrent_handle_conn_req_and_close_listener)
+{
+    m_reject_conn_request = true;
+
+    start_listen(concurrent_conn_request_and_listener_close_cb);
+    connect(*m_client, 0, client_disconnect_cb, nullptr);
+
+    while (!m_server_recv_req_cnt) {
+        std::this_thread::yield();
+    }
+
+    m_state |= TEST_STATE_SERVER_LISTENER_READY_CLOSE;
+    uct_listener_destroy(m_server->revoke_listener());
+    m_state |= TEST_STATE_SERVER_LISTENER_CLOSED;
+
+    wait_for_bits(&m_state, TEST_STATE_SERVER_REJECTED |
+                            TEST_STATE_CLIENT_GOT_REJECT);
+    EXPECT_TRUE(ucs_test_all_flags(m_state, (TEST_STATE_SERVER_REJECTED |
```

Comment:
1. extra ()
2. maybe when we use EXPECT_TRUE, could test each flag separately, instead of test_all_flags . it would help pinpoint the problem (in multiple places). same for EXPECT_FALSE.